### DISCUSSION TITLE
[Android] Camera plugin - access to photos on user phone

### DISCUSSION THREAD

**Discussion by hvma411** [ionic-team/capacitor#6389] [Q&A]
I'm creating some ionic capacitor app and I have problem with new Android phones. I have a screen with some button that allows user to set his profile picture. When user clicks button for the first time the app asks for permissions "Allow {APP_NAME} to access photos, videos, music and audio on this device?" and when user allow this permissions then it is still impossible to open gallery to choose photos.

I can see this error message in console: 

`I/Capacitor/Console: File: http://localhost/default-src_app_shared_shared_module_ts.js - Line 2195 - Msg: Error: Missing the following permissions in AndroidManifest.xml:
    android.permission.WRITE_EXTERNAL_STORAGE` 

I'm sure I have this permission in AndroidManifest.xml because I checked it a few times and it is there. I can't find anything that may help me to fix it.

My AndroidManifest.xml:

    <?xml version="1.0" encoding="utf-8"?>
    <manifest xmlns:android="http://schemas.android.com/apk/res/android"
    	xmlns:tools="http://schemas.android.com/tools"
        package="io.the.app">
    
        <application
            android:allowBackup="true"
            android:icon="@mipmap/ic_launcher"
            android:label="@string/app_name"
            android:roundIcon="@mipmap/ic_launcher_round"
            android:supportsRtl="true"
            android:theme="@style/AppTheme">
    		
         	<meta-data android:name="com.transistorsoft.locationmanager.license" android:value="" />
            <activity
                android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|smallestScreenSize|screenLayout|uiMode"
                android:name="io.the.app.MainActivity"
                android:label="@string/title_activity_main"
                android:theme="@style/AppTheme.NoActionBarLaunch"
                android:launchMode="singleTask"
                android:exported="true">
    
                <intent-filter>
                    <action android:name="android.intent.action.MAIN" />
                    <category android:name="android.intent.category.LAUNCHER" />
                </intent-filter>
    
            </activity>
    
            <provider
                android:name="androidx.core.content.FileProvider"
                android:authorities="${applicationId}.fileprovider"
                android:exported="false"
                android:grantUriPermissions="true">
                <meta-data
                    android:name="android.support.FILE_PROVIDER_PATHS"
                    android:resource="@xml/file_paths"></meta-data>
            </provider>
        </application>
    
        
    
        <uses-permission android:name="android.permission.INTERNET" />
    	<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    	<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    	<uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />
        <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
        <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
        <uses-permission android:name="android.permission.CAMERA" />
    	<uses-feature android:name="android.hardware.location.gps" />
    </manifest>

My capacitor doctor:

```
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 4.7.0
  @capacitor/core: 4.7.0
  @capacitor/android: 4.7.0
  @capacitor/ios: 4.7.0

Installed Dependencies:

  @capacitor/cli: 4.6.1
  @capacitor/core: 4.6.1
  @capacitor/android: 4.6.1
  @capacitor/ios: 4.6.1

[success] iOS looking great! 👌
[success] Android looking great! 👌
```

My camera plugin version:

`"@capacitor/camera": "^4.1.4",`

My function for adding photo looks like this:

    handleAddPhoto = async () => {
    	await Camera.checkPermissions().then((permissions) => {
    		if (permissions.photos !== 'granted') {
    			Camera.requestPermissions();
    		}
    	});
    	const photo = await Camera.getPhoto({
    		quality: 50,
    		resultType: CameraResultType.Base64,
    		source: CameraSource.Photos,
    	});
    	this.avatarSrc = this.sanitizeSrc(photo.base64String);
    	this.cdr.detectChanges();
    	this.handleSavePhoto('data:image/jpeg;base64,' + photo.base64String);
    };

**Comment by jcesarmobile** [comment]
hard to tell without the actual app to test, if the permission is there it should find it, but doesn't find it for some reason.

**Reply by hvma411** [reply]
Yup I know but this is company project so I can't share app code. Is there anything more needed to do besides requesting permissions and adding them to AndroidManifest.xml? Additional info: this problem appears only at new android versions, so for example android version 8 works fine

**Reply by jcesarmobile** [reply]
Camera plugin doesn’t work in android 13 if you use targetSDK 33, works fine if using targetSDK 32.
Other than that there are no known issues with other new android versions.

**Reply by hvma411** [reply]
Unfortunately changing targetSDK didn't change anything with my problem. I tried with target SDK 31 and 32 and nothing happend. My compiled app still don't see `<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />` .

**Reply by jcesarmobile** [reply]
so, you were using targetSDK 33?
try uninstalling the app as sometimes once you change the targetSDK the app will behave as the higher version you've set even if you go back to a previous version on already installed apps, but uninstalling and reinstalling usually fix that.

**Reply by hvma411** [reply]
No, by default there was targetSDK 32, so I tried to change it to 31 and then back to 32. I always uninstall app from device/simulator, clean build folder, build app again, check configuration, sync and then run app on device/simulator. I'm really confused what is going on with this problem

**Reply by jcesarmobile** [reply]
The problem I mentioned only happens when targeting SDK 33, not 32 or 31 (they are almost the same), and only on Android 13 devices, on other Android versions it should work.

**Reply by hvma411** [reply]
I'll try to create new empty ionic project with the same configuration as it is in my current project and then I'll create some simple screen with camera usage. Let's see what will happen
